### PR TITLE
Dockerize POC-Microfrontends and Import Map Deployer

### PR DESCRIPTION
**Pull request by drizzentic**
The PR introduces a script that utilizes single spa microfrontends server to host the import maps and build files for the apps within the monorepo.
The script generates an import map based on the packages in the monorepo. TB-388

### CODE REVIEW

**Comment by rcrichton** [comment]
Hi, I think this looks good but I'm struggling to understand exactly what it the script does and what the docker container ends up looking like. @drizzentic if you could help me understand that would be great.

**Comment by drizzentic** [comment]
Hi @rcrichton .
The script creates build for all the packages in the github monorepo and creates a docker image which has all the apps which is deployed to the QA server as a docker container.

File: packaging/import-maps-server/build.sh [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+            #Add package to the JSON file
+            cd "../../packaging/import-maps-server/" || exit
+            
+            jq --arg module_url "./libs/$package_name/dist/$file_name.js" ".imports += {\"$package_name\": \$module_url}" "$packages_json" >"$packages_json.tmp" && mv "$packages_json.tmp" "$packages_json"
+        )
+    fi
+done
+cd  packaging/import-maps-server/ || exit
+
+# echo "docker build -f Dockerfile-mf --build-arg WORK_DIR=$(pwd) --progress=plain --no-cache . -t $DOCKER_ORG_NAME/openhim-console-mf:$ENVIRONMENT_NAME --build-arg sourceDir=$(pwd)/packaging --build-arg libVersion=$PROJECT_UNIQUE_HASH_VERSION --build-arg baseImage=jembi/import-maps-mfe-server"
```
Comment (rcrichton) [inline packaging/import-maps-server/build.sh:55]: Minor: this debug message can probably be removed now

File: packages/openhim-core-api/src/jembi-openhim-core-api.ts [hunk 2]
```diff
@@ -76,5 +76,8 @@ export async function fetchAppsGroupedByCategory() {
 
 export async function addApp(app: App) {
   const response = await apiClient.post('/apps', app)
+
+
+  // insertt new app into the apps array. writere to to the too level of the json file
```
Comment (rcrichton) [inline packages/openhim-core-api/src/jembi-openhim-core-api.ts:81]: Minor: typo

File: packages/openhim-core-api/src/jembi-openhim-core-api.ts [hunk 3]
```diff
@@ -5,7 +5,7 @@ if running locally set REACT_APP_OPENHIM_API_BASE_URL environment variable to po
  **************************************************************************************************************** */
 
 const API_URL =
-  process.env.REACT_APP_OPENHIM_API_BASE_URL || 'https://localhost:8080/'
+  process.env.REACT_APP_OPENHIM_API_BASE_URL || 'https://13.245.11.170:8080/'
```
Comment (rcrichton) [inline packages/openhim-core-api/src/jembi-openhim-core-api.ts:8]: Should this not be reverted for local dev? If we want to provide QA details we should use the env var.

File: packages/legacy-app/package.json [hunk 4]
```diff
@@ -100,7 +100,7 @@
 		"clean": "rimraf dist",
 		"lint": "standard 'app/**/*.js'",
 		"lint:fix": "npm run lint -- --fix",
-		"test": "karma start --single-run",
+
```
Comment (rcrichton) [inline packages/legacy-app/package.json:103]: Why is this removed in the legacy app? Shouldn't we still be able to run it's tests?

**Review by rcrichton — CHANGES_REQUESTED** [review]
Looks good, a few comments.

File: packages/legacy-app/package.json (hunk 4 above)
Comment (drizzentic) [reply packages/legacy-app/package.json]: I have fixed the karma conf and some failing test and enabled this script to run

File: packages/openhim-core-api/src/jembi-openhim-core-api.ts (hunk 3 above)
Comment (drizzentic) [reply packages/openhim-core-api/src/jembi-openhim-core-api.ts]: This has been reverted and added to the build script

**Review by rcrichton — APPROVED** [review]
LGTM

**Pull request merged**